packages/agoric-cli/src/commands/wallet.js

```diff
@@ -0,0 +1,113 @@
+// @ts-check
+/* eslint-disable func-names */
+/* global fetch, process */
+import {
+  iterateLatest,
+  makeCastingSpec,
+  makeFollower,
+  makeLeaderFromRpcAddresses,
+} from '@agoric/casting';
+import { Command } from 'commander';
+import { fmtRecordOfLines, simplePurseBalances } from '../lib/format.js';
+import { simpleOffers } from '../lib/psm.js';
+import { makeRpcUtils, networkConfig } from '../lib/rpc.js';
+import { getWalletState } from '../lib/wallet.js';
+
+// Ambient types. Needed only for dev but this does a runtime import.
+import '@agoric/vats/src/core/types.js';
+import '@agoric/vats/src/types.js';
+import { makeLeaderOptions } from '../lib/casting.js';
+import { normalizeAddress } from '../lib/keys.js';
+
+export const makeWalletCommand = async () => {
+  const wallet = new Command('wallet').description('wallet commands');
+
+  wallet
+    .command('send')
+    .description('send a prepared offer')
+    .requiredOption(
+      '--from [address]',
+      'address literal or name',
+      normalizeAddress,
+    )
+    .requiredOption('--offer [filename]', 'path to file with prepared offer')
+    .action(function () {
+      const { from, offer } = this.opts();
+      const { chainName, rpcAddrs } = networkConfig;
+
+      const cmd = `agd --node=${rpcAddrs[0]} --chain-id=${chainName} --from=${from} tx swingset wallet-action --allow-spend "$(cat ${offer})"`;
+
+      process.stdout.write('Run this interactive command in shell:\n\n');
+      process.stdout.write(cmd);
+      process.stdout.write('\n');
+    });
+
+  wallet
+    .command('list')
+    .description('list all wallets in vstorage')
+    .action(async function () {
+      const { vstorage } = await makeRpcUtils({ fetch });
+      const wallets = await vstorage.keys('published.wallet');
```

Comment:
There could / should be a _lot_ of these in production use. I didn't implement paging in `vstorage.keys`. Some sort of iterator will probably be best in due course.

Reply:
yeah I figure when the list is too large we tackle that. having it here I hope will give us earlier warning